core/coreunix/add.go

```diff
@@ -153,8 +161,55 @@ func (params *Adder) PinRoot() error {
 	return params.node.Pinning.Flush()
 }
 
-func (params *Adder) Finalize(DAG dag.DAGService) (*dag.Node, error) {
-	return params.editor.Finalize(DAG)
+func (params *Adder) Finalize() (*dag.Node, error) {
+	root, err := params.mr.GetValue().GetNode()
+	if err != nil {
+		return nil, err
+	}
+
+	params.RootNode()
```

Comment:
why is this call here? since you dont capture the value it looks like it's causing some sideeffect, maybe comment?